### PR TITLE
#167398166 User should be able to view all articles

### PR DESCRIPTION
**Pull request by neelxie**
#### What does this PR do?

User is able to view all articles.

#### Description of Task to be completed?

- Retrieve all available articles
- Display all retrieved articles
- If no articles display appropriate message

#### How should this be manually tested?

- Git fetch this branch `ft-view-all-articles-167398166`
- Run the application with `npm start` and navigate to `localhost:8080/article`
- Without an internet connection, the response will be "No articles Available" while with a connection all articles will be displayed

#### Any background context you want to provide?

There was no provision of view all the articles now it is being provided.

#### What are the relevant pivotal tracker stories?

[#167398166](https://www.pivotaltracker.com/story/show/#167398166)

#### Screenshots:

My internet connection has failed to upload the images

### CODE REVIEW

File: src/Articles/components/AllArticles.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable react/forbid-prop-types */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prop-types */
```
Comment (jennizalwango) [inline src/Articles/components/AllArticles.jsx]: Please consider removing these Eslint disables

File: src/App.jsx [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
 export default function App() {
   return (
-    <Router>
+    <Provider store={store}>
+      <Router>
       <HeaderSection />
       <Switch>
         <Route path="/" exact component={Landing} />
         <Route path="/about" component={Landing} />
+        <Route path="/article" component={AllArticles} />
```
Comment (alexxsanya) [inline src/App.jsx]: @neelxie set the path to `/articles` instead of `/article`

File: src/Articles/components/AllArticles.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import React, { Component } from "react";
+import { Container } from "react-bootstrap";
+import PropTypes from "prop-types";
+import Article from "../../commons/components/Article";
+import { connect } from "react-redux";
+import getAllArticles from "../redux/actions/AllArticlesActions";
+import '../css/AllArticles.scss';
```
Comment (alexxsanya) [inline src/Articles/components/AllArticles.jsx]: @neelxie please consider changing the css folder name to uppercase like `CSS` to avoid rebase issues.

**Review by alexxsanya — CHANGES_REQUESTED** [review]

File: src/Articles/components/AllArticles.jsx (hunk 3 above)
Comment (neelxie) [reply src/Articles/components/AllArticles.jsx]: @alexxsanya thank you I have changed that.

File: src/App.jsx (hunk 2 above)
Comment (neelxie) [reply src/App.jsx]: @alexxsanya fixed, thank you.

File: src/Articles/components/AllArticles.jsx (hunk 1 above)
Comment (neelxie) [reply src/Articles/components/AllArticles.jsx]: @jennizalwango please noted the lines are on the develop as well, but I have removed them.

**Review by alexxsanya — CHANGES_REQUESTED** [review]
@neelxie kindly rebase your work and I should be able to approve then.

**Review by Emmanuelaaron — CHANGES_REQUESTED** [review]
Hi @neelxie,
Please fix travisc.
I can see the tests are failing

**Comment by neelxie** [comment]
> @neelxie kindly rebase your work and I should be able to approve then.

@alexxsanya I have rebased please approve.

**Comment by neelxie** [comment]
> Hi @neelxie,
> Please fix travisc.
> I can see the tests are failing

@Emmanuelaaron everything seems fine now, please recheck it.

**Review by alexxsanya — APPROVED** [review]

**Review by jennizalwango — APPROVED** [review]

**Review by HabibSentongo — APPROVED** [review]
Wow, this is good work. All functionality is working well.

**Review by Emmanuelaaron — APPROVED** [review]
All seems working well on my end

**Review by jennizalwango — APPROVED** [review]

**Review by jennizalwango — APPROVED** [review]

File: src/Articles/components/AllArticles.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  is_retrieving: PropTypes.bool
+};
+
+AllArticles.defaultProps = {
+  allArticles: [],
+  is_retrieving: false
+};
+
+export const mapStateToProps = state => {
+  const { is_retrieving, items } = state.allArticleReducer;
```
Comment (AGMETEOR) [inline src/Articles/components/AllArticles.jsx]: Please use the camelCase convention. Also, find anywhere else in your code where variables are not camelCase and change this

File: src/Articles/tests/components/AllArticles.test.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      mapStateToProps({
+        allArticleReducer: { is_retrieving: false, items: [] }
+      })
+    ).toEqual({ is_retrieving: false, items: [] });
+  });
+  it("match map state to dispatch", () => {
+    const dispatch = jest.fn();
+    expect(mapDispatchToProps(dispatch)).toHaveProperty("getArticles");
+  });
+});
```
Comment (AGMETEOR) [inline src/Articles/tests/components/AllArticles.test.jsx:43]: Also, test that indeed we call the getArticles function when component mounts and updates

**Review by AGMETEOR — CHANGES_REQUESTED** [review]

File: src/Articles/components/AllArticles.jsx (hunk 4 above)
Comment (neelxie) [reply src/Articles/components/AllArticles.jsx]: @AGMETEOR thank you, I have fixed that as well.

File: src/Articles/tests/components/AllArticles.test.jsx (hunk 5 above)
Comment (neelxie) [reply src/Articles/tests/components/AllArticles.test.jsx:43]: @AGMETEOR I have tested that and added a few more tests to increase coverage thank you.

**Review by AGMETEOR — APPROVED** [review]

**Pull request merged**